lib/charms/mongodb/v0/config_server_interface.py

```diff
@@ -105,8 +109,8 @@ def is_valid_mongos_integration(self) -> bool:
 
         return True
 
-    def _on_relation_changed(self, event) -> None:
-        """Handles providing mongos with KeyFile and hosts."""
+    def _on_database_requested(self, event: DatabaseRequestedEvent | RelationChangedEvent) -> None:
+        """Handles the database requested event, which is the first time we can write data."""
```

Comment:
We shouldn't do this. We should handle `DatabaseRequestedEvent` separate from others, performing its dedicated action (user creationg and credentials distribution)